pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,843 @@
+
+        Returns
+        -------
+        dict
+
+        """
+        props = self._app.design_properties
+        component_id = {}
+        user_defined_models = props["ModelSetup"]["GeometryCore"]["GeometryOperations"]["UserDefinedModels"][
+            "UserDefinedModel"
+        ]
+        if not isinstance(user_defined_models, list):
+            user_defined_models = [user_defined_models]
+        for component_defined in user_defined_models:
+            component_id[component_defined["ID"]] = component_defined["Attributes"]["Name"]
+
+        components_map = props["ArrayDefinition"]["ArrayObject"]["ComponentMap"]
+        components = [None] * len(components_map)
+        for comp in props["ArrayDefinition"]["ArrayObject"]["ComponentMap"]:
+            key, value = comp.split("=")
+            key = int(key.strip("'"))
+            value = int(value)
+            components[key - 1] = component_id[value]
+        array_props = OrderedDict()
+        array_props["component"] = components
+        array_props["active"] = props["ArrayDefinition"]["ArrayObject"]["Active"]["matrix"]
+        array_props["rotation"] = props["ArrayDefinition"]["ArrayObject"]["Rotation"]["matrix"]
+        array_props["cells"] = props["ArrayDefinition"]["ArrayObject"]["Cells"]["matrix"]
+        return array_props
+
+    @pyaedt_function_handler()
+    def _get_coordinate_system_id(self):
+        """Find coordinate system ID.
+
+        Returns
+        -------
+        int
+        """
+        id2name = {1: "Global"}
+        name2id = id2name
+        if self._app.design_properties and "ModelSetup" in self._app.design_properties:  # pragma: no cover
+            cs = self._app.design_properties["ModelSetup"]["GeometryCore"]["GeometryOperations"]["CoordinateSystems"]
+            for ds in cs:
+                try:
+                    if isinstance(cs[ds], (OrderedDict, dict)):
+                        name = cs[ds]["Attributes"]["Name"]
+                        cs_id = cs[ds]["ID"]
+                        id2name[cs_id] = name
+                    elif isinstance(cs[ds], list):
+                        for el in cs[ds]:
+                            name = el["Attributes"]["Name"]
+                            cs_id = el["ID"]
+                            id2name[cs_id] = name
+                except AttributeError:
+                    pass
+            name2id = {v: k for k, v in id2name.items()}
+        return name2id
+
+
+class CellArray(object):
+    """Manages object attributes for 3DComponent and User Defined Model.
+
+    Parameters
+    ----------
+    row : int
+        The row index of the cell.
+    col : int
+        The column index of the cell.
+    array_props : dict
+        Dictionary containing the properties of the array.
+    component_names : list
+        List of component names in the array.
+    array_obj : class:`pyaedt.modeler.cad.component_array.ComponentArray`
+        The instance of the array containing the cell.
```

Comment:
```suggestion
    row : int
        Row index of the cell.
    col : int
        Column index of the cell.
    array_props : dict
        Dictionary containing the properties of the array.
    component_names : list
        List of component names in the array.
    array_obj : class:`pyaedt.modeler.cad.component_array.ComponentArray`
        Instance of the array containing the cell.
```
We omit the leading article so that the first word of the description is an important one.